tests/core-addon/integration/core_addon.js

```diff
@@ -28,63 +28,154 @@ async function findAndAct(driver, element, action) {
+  await this.driver.wait(async () => {
+    return (await this.driver.getAllWindowHandles()).length === 2;
+  }, WAIT_FOR_PROPERTY);
+
+  // Selenium is still focused on the old tab, so switch to the new window handle.
+  const newTab = (await this.driver.getAllWindowHandles())[1];
+  await this.driver.switchTo().window(newTab);
+
+  // Switch back to web content context.
+  await this.driver.setContext(firefox.Context.CONTENT);
+
+  // New tab is focused, now on options page.
+  await this.driver.wait(
+    until.titleIs("Rally: Put your data to work for a better internet"),
+    WAIT_FOR_PROPERTY
+  );
+}
 
-    await this.driver.get(`file:///${__dirname}/index.html`);
-    await this.driver.wait(until.titleIs("Installation Test"), WAIT_FOR_PROPERTY);
-    await findAndAct(this.driver, By.id("install"), e => e.click());
+/**
+ * Join Rally.
+ * Assumes that the Rally core add-on is already installed, and
+ * focused on the options page.
+ *c
+ * Leaves the Selenium driver context set to CONTENT, and
+ * focused on the options page.
+ */
+async function joinRally(driver) {
+  this.driver = driver;
 
-    // switch to browser UI context, to interact with Firefox add-on install prompts.
-    await this.driver.setContext(firefox.Context.CHROME);
-    await findAndAct(this.driver, By.css(`[label="Add"]`), e => e.click());
-    await findAndAct(this.driver, By.css(`[label="Okay, Got It"]`), e => e.click());
+  await this.driver.wait(until.elementLocated(By.css("button")));
 
-    // Switch back to web content context.
-    await this.driver.setContext(firefox.Context.CONTENT);
+  // FIXME we need to use button IDs here so xpath is not needed...
+  // See https://github.com/mozilla-rally/core-addon/issues/244
+  await findAndAct(this.driver, By.xpath(`//button[text()="Get Started"]`), e => e.click());
+  await findAndAct(this.driver, By.xpath(`//button[text()="Accept & Participate"]`), e => e.click());
+  // TODO check that state is enrolled, see https://github.com/mozilla-rally/core-addon/issues/245
 
-    // We expect the extension to load its options page in a new tab.
-    await this.driver.wait(async () => {
-      return (await this.driver.getAllWindowHandles()).length === 2;
-    }, WAIT_FOR_PROPERTY);
+  await findAndAct(this.driver, By.xpath(`//button[text()="Save & Continue"]`), e => e.click());
 
-    // Selenium is still focused on the old tab, so switch to the new window handle.
-    const newTab = (await this.driver.getAllWindowHandles())[1];
-    await this.driver.switchTo().window(newTab);
+  // Switch back to web content context.
+  await this.driver.setContext(firefox.Context.CONTENT);
+}
 
-    // New tab is focused.
-    await this.driver.wait(
-      until.titleIs("Rally: Put your data to work for a better internet"),
-      WAIT_FOR_PROPERTY
-    );
+// A default set of test studies.
+// FIXME - this currently loads data from remote! We need to figure out how to host this locally.

```

Comment:
What data is loaded remotely? Do you mean the basic study?

Reply:
Yes, the local study and the icons. Everything with an `https://` link in here needs to be changed to either hit a local server, or we can see if `file://` works. I think it probably does.

Reply:
This is basically https://github.com/mozilla-rally/rally-core-addon/issues/253 which I think needs a followup issue?

Reply:
https://github.com/mozilla-rally/rally-core-addon/issues/319